src/Thorn_Calculus-Processes.thy

```diff
@@ -422,6 +436,10 @@ lemma adapted_after_general_parallel:
 
 (* FIXME: Check if we should also have a \<^theory_text>\<open>general_parallel_and_adapted\<close>. *)
 
+lemma family_uncurry_after_general_parallel:
+  shows "\<nabla> (\<lambda>a. \<Prod>v \<leftarrow> vs. \<P> v a) = \<Prod>v \<leftarrow> vs. \<nabla> (\<P> v)"
+  by (induction vs) (simp_all, metis)
```

Comment:
The use of `(simp_all, metis)` is slightly problematic, as what `metis` will find depends on what the automated proof method `simp_all` will leave over. In the lemma `adapted_after_general_parallel` just above, I explicitly stated what rewrite rules `simp_all` should use, and with this configuration it was even able to solve the induction cases on its own. When doing the same with this lemma, `simp_all` is enough as well, and the invocation of `metis` isn’t necessary anymore. Here’s the code that follows the proof of `adapted_after_general_parallel`:
```isabelle
  by
    (induction vs)
    (simp_all only: general_parallel.simps family_uncurry_after_stop family_uncurry_after_parallel)
```
If you think this is a good alternative, please amend this pull request accordingly.

Reply:
Fixed in ce11f14770a22de400a93333031708c5e2ab2382, please check.

Reply:
Perfect! Thanks.